sdk/src/main/java/io/dapr/client/domain/SubscriptionMetadata.java

```diff
@@ -0,0 +1,98 @@
+package io.dapr.client.domain;
+
+import java.util.List;
+
+public class SubscriptionMetadata {
+    private String topic;
+    private String pubsubname;
+    private String deadLetterTopic;
+    private List<RuleMetadata> rules;
+
+    public SubscriptionMetadata() {
+    }
+
+    public SubscriptionMetadata(String topic, String pubsubname, String deadLetterTopic, List<RuleMetadata> rules) {
+        this.topic = topic;
+        this.pubsubname = pubsubname;
+        this.deadLetterTopic = deadLetterTopic;
+        this.rules = rules;
```

Comment:
USe Collections.unmodifiableList()

Reply:
good point... now I see the other domain classes, thanks for pointing this out